### PR TITLE
Remove unused API and resolve naming ambiguities

### PR DESCRIPTION
**Pull request by raphw**
This PR makes a clear distinction between API and implementation, and cleans up a bit. For this:
- The internal `Tracer` API is removed.
- Implementation versions of API types are suffixed `Impl`.
- The `ServiceAwareTracer` API is consolidated with the `Tracer` API.
- Minor clean up of dependency imports.

### CODE REVIEW

File: apm-agent-plugins/apm-api-plugin/src/test/java/co/elastic/apm/api/AbstractSpanImplAccessor.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 public class AbstractSpanImplAccessor {
 
-    public static co.elastic.apm.agent.impl.transaction.Transaction accessTransaction(Transaction t) {
-        return (co.elastic.apm.agent.impl.transaction.Transaction) ((TransactionImpl) t).span;
+    public static co.elastic.apm.agent.impl.transaction.TransactionImpl accessTransaction(Transaction t) {
+        return (co.elastic.apm.agent.impl.transaction.TransactionImpl) ((TransactionImpl) t).span;
     }
```
Comment (SylvainJuge) [inline apm-agent-plugins/apm-api-plugin/src/test/java/co/elastic/apm/api/AbstractSpanImplAccessor.java:25]:
```suggestion
    public static TransactionImpl accessTransaction(Transaction t) {
        return (TransactionImpl) ((TransactionImpl) t).span;
    }
```

**Review by SylvainJuge — DISMISSED** [review]

File: apm-agent-plugins/apm-api-plugin/src/test/java/co/elastic/apm/api/AbstractSpanImplAccessor.java (hunk 1 above)
Comment (raphw) [reply apm-agent-plugins/apm-api-plugin/src/test/java/co/elastic/apm/api/AbstractSpanImplAccessor.java:25]: Those two are not the same `TransactionImpl`. It's `co.elastic.apm.agent.impl.transaction.TransactionImpl` and `co.elastic.apm.api.TransactionImpl`.

**Review by SylvainJuge — DISMISSED** [review]

**Review by SylvainJuge — DISMISSED** [review]

**Comment by jackshirazi** [comment]
note approved but failing

**Comment by raphw** [comment]
My bad. Non-trivial merge went too quick. It should work now!

**Review by SylvainJuge — APPROVED** [review]

**Comment by jackshirazi** [comment]
run docs-build

**Pull request merged**